aapcs64/aapcs64.rst

```diff
@@ -2026,6 +2026,21 @@ size\_t, ptrdiff\_t
 
 See `Arm C and C++ language mappings`_.
 
+Soft-float variant
+------------------
+
+For v8-R CPUs which do not have floating-point instructions or registers, the soft-float variant of the PCS can be used. Arm recommends that this is not used more widely, as most existing AArch64 implementations do have floating-point registers, and this variant is incompatible with the base variant.
+
+The soft-float variant is defined identically to the base variant, with these changes:
+
+* All floating-point types are passed as if they are an integer type with the same bit-width.
+
+* No types are considered to be HFAs or HVAs, these are treated as if they are non-homogeneous composite types.
```

Comment:
Does the implementation treat these as just structs of integers (of the appropriate size) for example a struct of two floats would be treated as if it were a struct of two integers.

From the specification
```
struct S {
  int x; // substitute for float
  int y; // substitute for float
};
```
Would be considered a homogenous aggregate, just not a homogenous floating point aggregate. I can't see anything in the argument passing rules that does anything special with this definition so it would behave as if a non-homogenous composite type. However for future changes would we prefer it to be defined as a homogenous aggregate or a non-homogenous composite?

If we do want to go down that route perhaps:
``Types that the base procedure call standard would consider HFAs or HVAs are treated as homogenous aggregates of signed Integral type with the same byte size and alignment."

Reply:
I agree that the original wording conflicts with the existing definition of `homogenous aggregate`, but I think that change could be confusing, since `homogenous aggregates of signed Integral type` isn't used anywhere else. Maybe it would be better to cut this down to just "No types are considered to be HFAs or HVAs"?

Reply:
I'd be happy with just "No types are considered to be HFAs or HVAs" I think that is sufficient for someone to work out what they need to do by following the rest of the rules in the doc and avoids a potentially confusing definition.